src/sinks/elasticsearch.rs

```diff
@@ -395,13 +396,39 @@ fn healthcheck(
 }
 
 fn signed_request(method: &str, uri: &Uri) -> SignedRequest {
-    let region = Region::Custom {
-        name: "custom".into(),
-        endpoint: uri.to_string(),
-    };
+    let region = derive_region(uri);
     SignedRequest::new(method, "es", &region, uri.path())
 }
 
+/// Derive a region name from a hostname
+fn derive_region(uri: &Uri) -> Region {
+    uri.host()
+        // Strip the standard domain name top level
+        .and_then(|hostname| {
+            if hostname.ends_with(".amazonaws.com") {
```

Comment:
Would it better to split on `.` and then just grab the second item? I doubt this will change any time soon, but `cn` provision suggests that this might be a better strategy.

Reply:
Perhaps. It would be simpler, but I get nervous about hard-coding assumptions like that. I'll change it.

Reply:
That's true. `..hostname.len() - 14` is also an assumption, so I'm not sure which one is less likely to change. Go with what you think is best.

Reply:
This may accidentally give false positives for non-AWS service host names, but usually those won't actually match an AWS region, so the `from_str` will still fail them.

Reply:
Note that `&hostname[..hostname.len() - 14]` is predicated on `hostname.ends_with(".amazonaws.com")` which is 14 characters long, so that's not the actual assumption. The assumption is that it should end with one of those two domain names.

Reply:
Got it. Go with what you think is best here.

Reply:
Besides maybe moving them to a constant that seems fine to me.